rmw_connextdds_common/src/common/rmw_context.cpp

```diff
@@ -181,13 +181,59 @@ rmw_context_impl_t::initialize_participant(const bool localhost_only)
 {
   RMW_CONNEXT_LOG_DEBUG("initializing DDS DomainParticipant")
 
+  this->localhost_only = localhost_only;
+
+  /* Lookup RMW_CONNEXT_ENV_ALLOW_TOPIC_QOS_PROFILES env variable.*/
+  const char * endpoint_qos_policy = nullptr;
+  const char * lookup_rc = rcutils_get_env(
+    RMW_CONNEXT_ENV_ENDPOINT_QOS_OVERRIDE_POLICY, &endpoint_qos_policy);
+
+  if (nullptr != lookup_rc || nullptr == endpoint_qos_policy) {
+    RMW_CONNEXT_LOG_ERROR_A_SET(
+      "failed to lookup from environment: "
+      "var=%s, "
+      "rc=%s ",
+      RMW_CONNEXT_ENV_ENDPOINT_QOS_OVERRIDE_POLICY,
+      lookup_rc)
+    return RMW_RET_ERROR;
+  }
+
+  this->endpoint_qos_override_policy = rmw_context_impl_t::endpoint_qos_override_policy_t::Always;
+  const char dds_topic_policy_prefix[] = "dds_topics: ";
+  const char never_policy[] = "never";
+  const char always_policy[] = "always";
+  if (
+    0 == strncmp(
+      endpoint_qos_policy, dds_topic_policy_prefix, sizeof(dds_topic_policy_prefix) - 1u))
+  {
+    this->endpoint_qos_override_policy =
+      rmw_context_impl_t::endpoint_qos_override_policy_t::DDSTopics;
+    try {
+      fprintf(stderr, "regex found: %s", &endpoint_qos_policy[sizeof(dds_topic_policy_prefix)]);
```

Comment:
Almost didn't catch this sneaky debug statement which we should remove.

Reply:
ups, fixed in https://github.com/rticommunity/rmw_connextdds/pull/7/commits/6d929153b72fa626f920228e91a45a24b8d72394